pandas/io/parsers.py

```diff
@@ -1274,7 +1258,22 @@ def _validate_parse_dates_arg(parse_dates):
 
 
 class ParserBase:
-    def __init__(self, kwds):
+    def __init__(self, src, kwds):
+
+        if isinstance(src, list):
+            self.handles = IOHandles(
+                handle=src, compression={"method": None}  # type: ignore[arg-type]
```

Comment:
how does this work?

Reply:
yeah, that isn't nice (especially from a typing perspective). The parsers seems to cope with objects implementing `__next__` (for example lists). I think a solution would be to have a new attribute for the parser (maybe `self.iterable : Optional[Iteratable]`). If it is None, use self.handles otherwise use the iterable?

Reply:
this branch is only used by `read_excel` and is only used for the `PythonParser`. I typed `src` and added comments about this special case.

I will try to fix this mess later this week.

Reply:
I think the 'best' solution is to temporarily silence mypy and put a list in `IOHandles`.  `PythonParser` (the only parser affected by that) had already a check whether the 'file handle' has `readline`, if not (for the list from `read_excel`) it treats it as raw data.